scripts/test/Abins/AbinsLoadDMOL3InternalsTest.py

```diff
@@ -0,0 +1,257 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2022 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from io import BytesIO
+import unittest
+
+import numpy as np
+from numpy.testing import assert_allclose
+
+from abins.input import DMOL3Loader
+from mantid.kernel import Atom
+
+
+class AbinsLoadDMOL3InternalsTest(unittest.TestCase):
+    """Check fragments of DMOL3 parser separately from whole file"""
+    def test_read_lattice_vectors(self):
+        file_without_vectors = BytesIO(b'No vectors in this file')
+        unit_cell = DMOL3Loader._read_lattice_vectors(
+            file_without_vectors)
+
+        assert_allclose(unit_cell, np.zeros((3, 3)))
+
+        file_with_vectors = BytesIO(b"""
+INCOOR, atomic coordinates in au (for archive):
+______________________________________________________________________>8
+
+$cell vectors
+             16.74108374131821    0.00000000000000    0.00000000000000
+             -8.37054187065910   14.49820380686420    0.00000000000000
+              0.00000000000000    0.00000000000000    9.52044021771771
+$coordinates
+F             2.22656413759532   -1.33383475061772   -1.80983568538814
+F             2.42243481736875    8.68442407988897    6.67097246055480
+F            -2.34542583215868   10.72867081788549    2.94943237944894
+""")
+        unit_cell = DMOL3Loader._read_lattice_vectors(file_with_vectors)
+
+        assert_allclose(unit_cell,
+                        np.array([[ 8.859     ,  0.        ,  0.        ],
+                                  [-4.4295    ,  7.67211905,  0.        ],
+                                  [ 0.        ,  0.        ,  5.038     ]]))
+
+    def test_read_positions(self):
+        file_with_coords_end = BytesIO(b"""
+             -8.37054187065910   14.49820380686420    0.00000000000000
+              0.00000000000000    0.00000000000000    9.52044021771771
+$coordinates
+F             2.22656413759532   -1.33383475061772   -1.80983568538814
+F             2.42243481736875    8.68442407988897    6.67097246055480
+Na           -2.34542583215868   10.72867081788549    2.94943237944894
+Si            0.04185270935330    2.59517848080392   -1.80983568538814
+$end
+DUMMY CONTENT
+""")
+        atoms = DMOL3Loader._read_atomic_coordinates(file_with_coords_end)
+
+        assert_allclose(
+            [atom['coord'] for _, atom in atoms.items()],
+            np.array([[2.22656413759532, -1.33383475061772, -1.80983568538814],
+                      [2.42243481736875, 8.68442407988897, 6.67097246055480],
+                      [-2.34542583215868, 10.72867081788549, 2.94943237944894],
+                      [0.04185270935330, 2.59517848080392, -1.80983568538814]]
+                     ) * 0.52917721067  # Convert Bohr to Angstrom
```

Comment:
we should import the conversion factor (forces the same precision and is cleaner). Otherwise we should just report the values in Angstroms.

Reply:
I worry that this will miss the case that something goes wrong with `abins.constants`, but I guess that will still come up in the numeric tests!

Reply:
either that or do a test to check that the constant has a specific value

Reply:
Gone with import: meaningful names > comments